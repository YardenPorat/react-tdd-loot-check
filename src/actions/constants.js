export const SET_BALANCE = 'SET_BALANCE';
export const ADD_TO_BALANCE = 'ADD_TO_BALANCE';
export const WITHDRAW_BALANCE = 'WITHDRAW_BALANCE';
export const FETCH_BITCOIN = 'FETCH_BITCOIN';
