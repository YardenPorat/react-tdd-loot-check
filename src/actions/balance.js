import * as constants from './constants';

export const setBalance = balance => {
  return {
    type: constants.SET_BALANCE,
    balance,
  };
};

export const deposit = amount => {
  return {
    type: constants.ADD_TO_BALANCE,
    amount,
  };
};

export const withdraw = withdrawal => {
  return {
    type: constants.WITHDRAW_BALANCE,
    withdrawal,
  };
};
