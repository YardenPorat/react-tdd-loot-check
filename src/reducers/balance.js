import * as constants from '../actions/constants';

export const balanceReducer = (state = 0, action) => {
  let balance;

  switch (action.type) {
    case constants.SET_BALANCE:
      balance = action.balance;
      break;
    case constants.ADD_TO_BALANCE:
      balance = action.amount + state;
      break;
    case constants.WITHDRAW_BALANCE:
      balance = state - action.withdrawal;
      break;
    default:
      balance = state;
  }
  return balance;
};
