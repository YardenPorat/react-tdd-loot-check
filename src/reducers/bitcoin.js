import { FETCH_BITCOIN } from '../actions/constants';

export const bitcoinReducer = (state = {}, action) => {
  switch (action.type) {
    case FETCH_BITCOIN:
      return action.bitcoin;
    default:
      return state;
  }
};
