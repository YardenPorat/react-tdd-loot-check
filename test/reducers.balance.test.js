import { balanceReducer } from '../src/reducers/balance';
import * as constants from '../src/actions/constants';
import { expect } from 'chai';
import { describe } from 'mocha';

describe('Balance Reducer', () => {
  describe('when initializing', () => {
    const balance = 10;

    it('sets a balance', () => {
      const result = balanceReducer(undefined, {
        type: constants.SET_BALANCE,
        balance,
      });
      expect(result).to.eql(balance);
    });
  });

  it('should add to the balance', () => {
    const amount = 10;
    const initialState = 5;
    const result = balanceReducer(initialState, {
      type: constants.ADD_TO_BALANCE,
      amount,
    });
    expect(result).to.eql(amount + initialState);
  });

  it('Creates an action to withdraw from the balance', () => {
    const withdrawal = 3;
    const initialState = 13;

    const result = balanceReducer(initialState, {
      type: constants.WITHDRAW_BALANCE,
      withdrawal,
    });
    expect(result).to.eql(initialState - withdrawal);
  });
});
