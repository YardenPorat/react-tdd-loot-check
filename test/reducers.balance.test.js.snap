// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Balance Reducer Creates an action to withdraw from the balance 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Loot Check
        </h2>,
        <UNDEFINED />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loot Check",
        },
        "ref": null,
        "rendered": "Loot Check",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Loot Check
          </h2>,
          <UNDEFINED />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loot Check",
          },
          "ref": null,
          "rendered": "Loot Check",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Balance Reducer Creates an action to withdraw from the balance 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
    deposit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
        >
          Wallet balance: 
          20
        </h3>,
        <input
          className="input-wallet"
          onChange={[Function]}
          type="text"
        />,
        <button
          className="btn-deposit"
          onClick={undefined}
        >
          Deposit
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet balance: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet balance: ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-wallet",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Deposit",
          "className": "btn-deposit",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Deposit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
          >
            Wallet balance: 
            20
          </h3>,
          <input
            className="input-wallet"
            onChange={[Function]}
            type="text"
          />,
          <button
            className="btn-deposit"
            onClick={undefined}
          >
            Deposit
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet balance: ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Wallet balance: ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-wallet",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Deposit",
            "className": "btn-deposit",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Deposit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`Balance Reducer Creates an action to withdraw from the balance 3`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Loot Check
        </h2>,
        <UNDEFINED />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loot Check",
        },
        "ref": null,
        "rendered": "Loot Check",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "type": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Loot Check
          </h2>,
          <UNDEFINED />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loot Check",
          },
          "ref": null,
          "rendered": "Loot Check",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "type": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`Balance Reducer Creates an action to withdraw from the balance 4`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Wallet
    balance={20}
    deposit={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3
          className="balance"
        >
          Wallet balance: 
          20
        </h3>,
        <input
          className="input-wallet"
          onChange={[Function]}
          type="text"
        />,
        <button
          className="btn-deposit"
          onClick={undefined}
        >
          Deposit
        </button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Wallet balance: ",
            20,
          ],
          "className": "balance",
        },
        "ref": null,
        "rendered": Array [
          "Wallet balance: ",
          20,
        ],
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "input-wallet",
          "onChange": [Function],
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Deposit",
          "className": "btn-deposit",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Deposit",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3
            className="balance"
          >
            Wallet balance: 
            20
          </h3>,
          <input
            className="input-wallet"
            onChange={[Function]}
            type="text"
          />,
          <button
            className="btn-deposit"
            onClick={undefined}
          >
            Deposit
          </button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Wallet balance: ",
              20,
            ],
            "className": "balance",
          },
          "ref": null,
          "rendered": Array [
            "Wallet balance: ",
            20,
          ],
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "input-wallet",
            "onChange": [Function],
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Deposit",
            "className": "btn-deposit",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Deposit",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
